### PR TITLE
Refactor main

### PR DESCRIPTION
**Pull request by NilsHullegien**
Refactoring of the Main class.
This class was appended with a VRConfigurer class (which configures the VR settings) and an InputHandler class (which handles the keyInputs when in game). The Main class still contains the main method of the game.

### CODE REVIEW

**Comment by marcelkuipers** [comment]
Maybe you can create a new package for the new classes. For the rest it looks good.

File: ContextDesktop/src/main/java/com/github/migi_1/Context/InputHandler.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        return up;
+    }
+
+    public boolean isDown() {
+        return down;
+    }
+
+    public ActionListener getActionListener() {
+        return actionListener;
+    }
```
Comment (LMiljak) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/InputHandler.java]: Missing some Javadoc in this class.

File: ContextDesktop/src/test/java/com/github/migi_1/Context/TestInputHandler.java [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+    @Test
+    public void testCheckSteerReset() {
+        Mockito.verify(envState, Mockito.never()).steer(Mockito.anyFloat());
+        inputHandler.getActionListener().onAction("steer_left", true, 0f);
+        Mockito.verify(envState, Mockito.never()).steer(0.0f);
+        inputHandler.getActionListener().onAction("steer_right", true, 0f);
+        Mockito.verify(envState, Mockito.never()).steer(0.0f);
+        inputHandler.getActionListener().onAction("steer_right", false, 0f);
+        Mockito.verify(envState).steer(1.0f);
+    }
```
Comment (LMiljak) [inline ContextDesktop/src/test/java/com/github/migi_1/Context/TestInputHandler.java:286]: Missing a lot of Javadoc in this class

File: ContextDesktop/src/test/java/com/github/migi_1/Context/TestMain.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        main = new Main();
+//        main.setUpMain();
+//        main.simpleInitApp();
+    }
+
+    @Test
+    public void simpleInitTest() {
+//        Mockito.verify(settings).setTitle(Mockito.anyString());
+//        Mockito.verify(settings).setResolution(Mockito.anyInt(), Mockito.anyInt());
+    }
```
Comment (LMiljak) [inline ContextDesktop/src/test/java/com/github/migi_1/Context/TestMain.java]: Does test does nothing.
Also, Javadoc.

File: ContextDesktop/src/test/java/com/github/migi_1/Context/TestVRConfigurer.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        main = Mockito.mock(Main.class);
+    }
+
+    @Test
+    public void configureVRTest() {
+        VRConfigurer.configureVR(main);
+        Mockito.verify(main, Mockito.times(8)).preconfigureVRApp(Mockito.any(), Mockito.anyBoolean());
+        Mockito.verify(main).preconfigureFrustrumNearFar(Mockito.anyFloat(), Mockito.anyFloat());
+    }
+
```
Comment (LMiljak) [inline ContextDesktop/src/test/java/com/github/migi_1/Context/TestVRConfigurer.java:37]: Javadoc

**Comment by LMiljak** [comment]
Fix Checkstyle violations please

**Comment by NilsHullegien** [comment]
Closing the PR to fix the comments.

**Comment by NilsHullegien** [comment]
Added javadoc, the classes are all moved into a .main package. Reopening PR

**Comment by NilsHullegien** [comment]
closing pr to fix merge conflicts

File: ContextDesktop/src/main/java/com/github/migi_1/Context/main/InputHandler.java [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+                            break;
+                        default: //Do nothing when an unknown button is pressed.
+                    }
+                }
+                checkSteering(name, keyPressed);
+            }
+        };
+        addListeners();
+    }
+
```
Comment (dVoorhout) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/main/InputHandler.java:72]: rather long method, could you split up this method?

**Comment by NilsHullegien** [comment]
Split up the method, good idea @indisputable!
Reopening PR.

**Comment by LMiljak** [comment]
I approve

**Pull request merged**